addons/web_editor/controllers/main.py

```diff
@@ -301,130 +301,164 @@ def get_image_info(self, image_id=None, xml_id=None):
+    def load_resources(self, file_type, views, bundles_restriction, only_user_custom_files):
+        files_data = []
+        resources_type_info = {'t_call_assets_attribute': 't-js', 'mimetype': 'text/javascript'}
+        if file_type == 'scss':
+            resources_type_info = {'t_call_assets_attribute': 't-css', 'mimetype': 'text/scss'}
+
+        # Compile regex outside of the loop
+        # This will used to exclude library scss files from the result
+        excluded_url_matcher = re.compile("^(.+/lib/.+)|(.+import_bootstrap.+\.scss)$")
+
+        # Load already customized files attachments
+        custom_url = self._make_custom_scss_or_js_file_url("%%.%%", "%%")
+        custom_attachments = self.get_custom_attachment(custom_url, op='=like')
+
+        custom_files = [
+            '/website/static/src/scss/user_custom_bootstrap_overridden.scss',
+            '/website/static/src/scss/user_custom_rules.scss',
+            '/website/static/src/js/user_custom_javascript.js'
+        ]
+
+        # First check the t-call-assets used in the related views
+        url_infos = dict()
+        for v in views:
+            for asset_call_node in etree.fromstring(v["arch"]).xpath("//t[@t-call-assets]"):
+                if asset_call_node.get(resources_type_info['t_call_assets_attribute']) == "false":
+                    continue
+                asset_name = asset_call_node.get("t-call-assets")
+
+                # Loop through bundle files to search for file info
+                _files_data = []
+                for file_info in request.env["ir.qweb"]._get_asset_content(asset_name, {})[0]:
+                    if file_info["atype"] != resources_type_info['mimetype']:
+                        continue
+                    url = file_info["url"]
+
+                    # Exclude library files (see regex above)
+                    if excluded_url_matcher.match(url):
+                        continue
+
+                    # Set files URL filters
+                    url_from_custom_files = url in custom_files
+
+                    # Check if the file is customized and get bundle/path info
+                    scss_or_js_file_data = self._match_scss_or_js_file_url(url)
+                    if not scss_or_js_file_data:
+                        continue
+                    file_is_customized = scss_or_js_file_data['customized']
+
+                    # Save info according to the filter (arch will be fetched later)
+                    url_infos[url] = scss_or_js_file_data
+
+                    if file_type == 'scss':
+                        if only_user_custom_files:
+                            if url_from_custom_files or file_is_customized:
+                                _files_data.append(url)
+                        else:
+                            _files_data.append(url)
+                    else:
+                        if url_from_custom_files or file_is_customized:
+                            _files_data.append(url)
```

Comment:
```python
if A:
    if B:
        if C:
            do()
    else:
        do()
...
...
```

=>

```python
if A and B or ...:
    do()
```

Reply:
I'm stuck on this one. I understand and that was the direction I took at first, but still I can't get the right result..

Reply:
This is just about regrouping the condition in one expression. There is A = `file_type == 'scss'`, B = `only_user_custom_files` and C = `url_from_custom_files or file_is_customized` and we do `.append(url)` only when `f(A, B, C)` is true. If your current code is right, you just have to find that `f(A, B, C)`

Reply:
Ok then I guess something is wrong in my code or it should have worked.

Reply:
Did you find the solution ? I would like to send this in india today as my logo task depends too much on this.

Reply:
No. The only solution with the same result is : 
```
if file_type == 'scss':
    if (only_user_custom_files and url_from_custom_files) or file_is_customized:
        files_data.append(url)
    if not only_user_custom_files:
        files_data.append(url)
else:
    if url_from_custom_files or file_is_customized:
        files_data.append(url)
```
which is pretty much the same code... I'm still looking for a better solution but still... I get a wrong result :disappointed: Another solution with much simpler is working until filter is set to `Load All SCSS files`, so no.

Reply:
Ok, push your work without this part, I will do it myself and show you

Reply:
Ok, sorry for this :tired_face: 

_note à moi même_
```
if file_type == 'scss' and (only_user_custom_files and url_from_custom_files or file_is_customized):
    files_data.append(url)
elif file_type == 'js' and (only_user_custom_files and url_from_custom_files or file_is_customized):
    files_data.append(url)
else:
    if not only_user_custom_files:
        files_data.append(url)
```